repose-aggregator/components/filters/saml-policy-translation-filter/src/main/scala/org/openrepose/filters/samlpolicy/SamlPolicyTranslationFilter.scala

```diff
@@ -242,6 +268,47 @@ class SamlPolicyTranslationFilter @Inject()(configurationService: ConfigurationS
                           })
     }
 
+    try {
+      // Create a DOM XMLSignatureFactory that will be used to
+      // generate the enveloped signature.
+      xmlSignatureFactory = XMLSignatureFactory.getInstance("DOM")
+      // Create a Reference to the enveloped document (in this case,
+      // you are signing the whole document, so a URI of "" signifies
+      // that, and also specify the SHA1 digest algorithm and
+      // the ENVELOPED Transform.
+      val ref = xmlSignatureFactory.newReference(
+        "",
+        xmlSignatureFactory.newDigestMethod(DigestMethod.SHA1, null),
+        Collections.singletonList(xmlSignatureFactory.newTransform(Transform.ENVELOPED, null.asInstanceOf[TransformParameterSpec])),
+        null,
+        null)
+      // Create the SignedInfo.
+      signedInfo = xmlSignatureFactory.newSignedInfo(xmlSignatureFactory.newCanonicalizationMethod(
+        CanonicalizationMethod.INCLUSIVE,
+        null.asInstanceOf[C14NMethodParameterSpec]),
+        xmlSignatureFactory.newSignatureMethod(SignatureMethod.RSA_SHA1, null),
+        Collections.singletonList(ref))
+      // Load the KeyStore and get the signing key and certificate.
+      val signatureCredentials = newConfiguration.getSignatureCredentials
+      val keyStoreFilename = FileUtilities.guardedAbsoluteFile(configRoot, signatureCredentials.getKeystoreFilename).getAbsolutePath
+      logger.debug("Attempting to load keystore located at: {}", keyStoreFilename)
+      val keyStore = KeyStore.getInstance("JKS")
+      keyStore.load(new FileInputStream(keyStoreFilename), signatureCredentials.getKeystorePassword.toCharArray)
+      keyEntry = keyStore.getEntry(signatureCredentials.getKeyName, new KeyStore.PasswordProtection(signatureCredentials.getKeyPassword.toCharArray)).asInstanceOf[KeyStore.PrivateKeyEntry]
+      val x509Certificate = keyEntry.getCertificate.asInstanceOf[X509Certificate]
+      // Create the KeyInfo containing the X509Data.
+      val keyInfoFactory = xmlSignatureFactory.getKeyInfoFactory
+      val x509Data = keyInfoFactory.newX509Data(List(x509Certificate.getSubjectX500Principal.getName, x509Certificate).asJava)
+      keyInfo = keyInfoFactory.newKeyInfo(Collections.singletonList(x509Data))
+    } catch {
+      case e @ (_ : NoSuchMechanismException |
+                _ : ClassCastException |
+                _ : IllegalArgumentException |
+                _ : GeneralSecurityException |
+                _ : KeyStoreException |
+                _ : IOException) => throw new UpdateFailedException("Failed to load the signing credentials.", e)
```

Comment:
Better, but damn is that a lot of things. Oh well what can you do when dealing with garbage apis.